### DISCUSSION TITLE
How to grab cache by id when passing when extracting ssrCache?

### DISCUSSION THREAD

**Discussion by iamshadmirza** [urql-graphql/urql#3225] [Q&A] [answered]
I'm using a query which can be accessed by multiple users for their specific profile page.
I'm getting cache by `ssrCache.extractData()` and passing it as props from `getServerSideProps`.

This results in passing cache of all the users whose profile's have been visited recently.
I want to be able to grab the cache by username or id of user and pass only that via props.

I have specified key using `keys` prop in `cacheExchange`
```
keys: {
  User: (data) => data.id
}
```
I was expecting this `id` to become the key in `ssrCache` which I can use later but the keys are still coming as random id `3320962404`.

What would be the right approach to go here?

**Comment by JoviDeCroock** [comment, marked as answer]
Are you perhaps creating the `ssrExchange`/`urqlClient`/`cacheExchange` outside of the `getServerSideProps`, in doing so you would re-use these instances which is most likely not desirable.

You also can just use the simplest urql-client in `getServerSideProps` something like `exchanges: [ssr, fetch]` as you most likely don't want to share caches among users

**Reply by iamshadmirza** [reply]
Yeah, this sounds like the correct way. I would still want to know how does keys work and if it’s possible to grab a cache object by key.

I’m seeing this as hotfix since the approach you suggested will require a bigger refactor and more time.

thanks for replying

**Reply by JoviDeCroock** [reply]
the ssr and cache-exchange have isolated functionality, so yes you will key by what you return in the normalized-cache but the ssr works agnostic of what cache you are using as it is its own cache

**Reply by iamshadmirza** [reply]
Alright, thanks for the help Jovi
